### ISSUE TITLE
Super eyedropper

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#6269]
### Why this would be helpful

Scratch's built-in eyedropper tool can only pick colors from the costume editor canvas or the stage if you have that addon enabled. But it can also be useful to be able to quickly pick any color on your screen and insert it into a costume in Scratch.

Alternatives exist - the popup displayed by the Hex color picker addon comes with an eyedropper tool that can pick any color on the screen. You can also use eyedropper tools in a desktop program or browser extension. But what if this could be done with two clicks instead of three or more?

### My suggestion

It's super easy to invoke the browser's color picker:
```js
const eyeDropper = new EyeDropper
eyeDropper.open().then((result) => console.log(result.sRGBHex))
```
so I thought it might be possible to turn the "Pick colors on stage with eyedropper" addon into an addon called "Super eyedropper" that adds the ability to pick a color from anywhere on the screen, including other windows.

**Comment by CST1229** [comment]
That would be Chromium-only, though:
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/99d20f5e-95eb-42da-9007-d54087532c3b)

**Comment by DNin01** [comment]
If only Chrome supports that, could we make this work by splitting the `pick-colors-from-stage` addon in two, one shown to Chrome users and one shown to Firefox users? Or could there be an option to switch between selecting only colors from the stage or canvas and picking from anywhere on the screen that would be hidden for browsers that don't support it?

**Comment by WorldLanguages** [comment]
I'd say we extend the `Pick colors on stage with eyedropper` addon to automatically use this if the EyeDropper API if available. Are there any downsides?

**Comment by DNin01** [comment]
> I'd say we extend the `Pick colors on stage with eyedropper` addon to automatically use this if the EyeDropper API if available. Are there any downsides?

We could probably still call it "Super eyedropper", but the description might be a little more complicated.

**Comment by CST1229** [comment]
> I'd say we extend the `Pick colors on stage with eyedropper` addon to automatically use this if the EyeDropper API if available. Are there any downsides?

I assume it will pick anti-aliasing colors and the paint editor checkerboard (while Scratch's eyedropper doesn't).

**Comment by WorldLanguages** [comment]
> I assume it will pick anti-aliasing colors and the paint editor checkerboard (while Scratch's eyedropper doesn't).

Yeah, that's a problem. Also, the browser's EyeDropper API probably won't let us activate it unless the user recently clicked somewhere in the page. So that means we can't transform the vanilla eyedropper to the browser one automatically when the cursor leaves an area of the editor, which would probably be ideal.

**Comment by DNin01** [comment]
> > I assume it will pick anti-aliasing colors and the paint editor checkerboard (while Scratch's eyedropper doesn't).
> 
> Yeah, that's a problem. Also, the browser's EyeDropper API probably won't let us activate it unless the user recently clicked somewhere in the page. So that means we can't transform the vanilla eyedropper to the browser one automatically when the cursor leaves an area of the editor, which would probably be ideal.

Should we just add a separate "super eyedropper" button, then?

**Comment by WorldLanguages** [comment]
> Should we just add a separate "super eyedropper" button, then?

Yeah, it could be named `pick color from different window` or something like that. And we just hide the button if the browser doesn't support this API.

**Comment by BroJac5246** [comment]
Let me ask the obvious question... why?

**Comment by WorldLanguages** [comment]
Why what...?

**Comment by BroJac5246** [comment]
> Why what...?

Why is this necessary? What's an example use case? Is it worth the effort, especially considering everything that's been brought up?

**Comment by WorldLanguages** [comment]
> What's an example use case?

Picking colors from other windows/monitors

**Comment by DNin01** [comment]
> > What's an example use case?
> 
> 
> 
> Picking colors from other windows/monitors

I think they're asking when we would ever need to pick colors from other windows.

So far, I've only used it to pick colors from Scratch (like the new purple and block colors) so I can insert them into a project.

**Comment by BroJac5246** [comment]
> > > What's an example use case?
> > 
> > 
> > 
> > Picking colors from other windows/monitors
> 
> I think they're asking when we would ever need to pick colors from other windows.
> 
> So far, I've only used it to pick colors from Scratch (like the new purple and block colors) so I can insert them into a project.

Yes, that's what I'm asking.

My point is, is it worth it? The block colors are a niche case that would likely be better of just using a regular color picker browser extension.

**Comment by Secret-chest** [comment]
What if you have a palette on the other window to pick from?

**Comment by BroJac5246** [comment]
> What if you have a palette on the other window to pick from?

Copy/paste the hex codes.

**Comment by Secret-chest** [comment]
> What if you have a palette on the other window to pick from?
> 
> Copy/paste the hex codes.

We're trying to avoid that.

**Comment by DNin01** [comment]
Thinking about how much someone would _actually_ use this, I'd say this is too complicated to be worth it. You already have access to this kind of eyedropper in the Hex color picker addon, which is what I use whenever I need to pick colors from other windows.

On the other hand, something like this addon would make this feature easier to find for anyone who needs it, but those users would probably just find a workaround.

Maybe we could somehow reference Hex color picker from `pick-colors-from-stage` (with #5937)?

**Comment by DNin01** [comment]
> > What if you have a palette on the other window to pick from?
> 
> Copy/paste the hex codes.

I think that would be a reasonable way to transfer colors from a palette on another window to Scratch with the use of Hex color picker.

**Comment by Samq64** [comment]
> You already have access to this kind of eyedropper in the Hex color picker addon.

I think we should mention clicking the colour opens the system's colour picker in that addon's description.

**Comment by WorldLanguages** [comment]
Firefox doesn't support this new API yet so nothing we can do there.

**Comment by WorldLanguages** [comment]
Somewhat? related feedback:

> Please add a faster way to use the eyedropper tool. It would save lots of time.

Not sure what they mean. Their Scratch username is billybot21

**Comment by CST1229** [comment]
> Somewhat? related feedback:
> 
> > Please add a faster way to use the eyedropper tool. It would save lots of time.
> 
> Not sure what they mean. Their Scratch username is billybot21

They might just be referring to an eyedropper shortcut (for example, in Aseprite you can just Alt+click to eyedrop instantly).
